src/jwplayer/utils/useAdComplete.ts

```diff
@@ -1,16 +1,16 @@
 // Source: https://usehooks.com/useOnScreen/
 import { useState, useEffect } from 'react';
-import { communicationService, ofType } from 'jwplayer/utils/communication';
-import { race, timer } from 'rxjs';
-import { first } from 'rxjs/operators';
+import { getCommunicationService } from 'jwplayer/utils/communication';
 import { recordVideoEvent, VIDEO_RECORD_EVENTS } from 'jwplayer/utils/videoTimingEvents';
 
 export default function useAdComplete(): boolean {
 	const [adComplete, setAdComplete] = useState(false);
+	const communicationService = getCommunicationService();
 
 	useEffect(() => {
 		recordVideoEvent(VIDEO_RECORD_EVENTS.JW_PLAYER_AD_ENG_OPT_IN_LISTEN_START);
-		communicationService.action$.pipe(ofType('[AdEngine OptIn] set opt in'), first()).subscribe(() => {
+		communicationService.on('[AdEngine OptIn] set opt in', () => {
+			console.log(' ---- [AdEngine OptIn] set opt in ---- ');
```

Comment:
Let's remove `console.log()`, so we don't add noise in the dev console. Or replace it with `console.debug()` so it doesn't show up in a regular dev tools console (only when someone filters for all the messages). Ultimately, I'd like to add there one more layer and push debug info only when we have a cookie or query-string 🤔 But `console.log()` is a no-no from me ;-)

Reply:
A Debug query string is a good idea